src/test/java/resources/application.properties

```diff
@@ -0,0 +1,5 @@
+spring.datasource.url=jdbc:h2:mem:book
```

Comment:
resources folder should be placed at `src/test/resources`, but you have `src/test/java/resources/`